src/utils/scripts.ts

```diff
@@ -4,6 +4,67 @@ import colors from "colors";
 import _ from "lodash";
 import { Logger } from "tslog";
 import { execaCommand } from "execa";
+import { debugLogger, log } from "./logging.js";
+
+export async function expandVariablesFromScript(
+    scripts: string | string[] | undefined,
+    functions:
+        | {
+              name: string;
+              url: string;
+          }[]
+        | undefined,
+): Promise<string | string[] | undefined> {
+    if (!scripts) {
+        return;
+    }
+
+    if (!Array.isArray(scripts)) {
+        scripts = [scripts];
+    }
+
+    const expandedScripts: string[] | string = [];
+
+    for (let script of scripts) {
+        const regex = /\$\{\{[a-zA-Z0-9-]+\}\}/g;
+        const matches = script.match(regex);
+        const prefix = "${{";
+        const suffix = "}}";
+
+        if (matches) {
+            for (const match of matches) {
```

Comment:
I would avoid extra indentation

```suggestion
        for (const match of matches || []) {
```

Reply:
Agree. Done